tools/toolchains/mbed_toolchain.py

```diff
@@ -1097,11 +1097,7 @@ def check_c_lib_supported(self, target, toolchain):
         target.c_lib is modified to have the lowercased string of its original string.
         This is done to be case insensitive when validating.
         """
-        if  hasattr(target, "default_lib"):
-            raise NotSupportedException(
-                   "target.default_lib is no longer supported, please use target.c_lib for C library selection."
-                )
-        if  hasattr(target, "c_lib"):
+        if  hasattr(target, "default_lib") or hasattr(target, "c_lib"):
```

Comment:
Looks like this allows default_lib to be specified without failing, but the code doesn't appear to do anything with this attribute here. I don't know whether the code handling default_lib has been removed, or that it still exists somewhere in the codebase and remains unaltered. Needs checking...

Reply:
Yes, this obviously won't work as intended. Will rework.